libraries/lib-wave-track/Sequence.cpp

```diff
@@ -1114,6 +1114,24 @@ bool Sequence::Read(samplePtr buffer, sampleFormat format,
    return true;
 }
 
+AudioSegmentSampleView
+Sequence::GetFloatSampleView(sampleCount start, size_t length) const
+{
+   assert(start < mNumSamples);
+   length = std::min(sampleCount { length }, mNumSamples - start).as_size_t();
```

Comment:
Note `as_size_t` contains an assertion that you are not narrowing, so I like it better than the unchecked narrowing.

Reply:
`limitSampleBufferSize` might also express this

Reply:
That function also guards against the case that the `sampleCount` is negative.  That can happen because the type is also meant to expressed signed differences of positions in audio streams.

Reply:
`limitSampleBufferSize` is handy :thumbs-up: